core/src/test/java/io/neow3j/protocol/rx/JsonRpc2_0RxTest.java

```diff
@@ -0,0 +1,296 @@
+package io.neow3j.protocol.rx;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.neow3j.protocol.Neow3j;
+import io.neow3j.protocol.Neow3jService;
+import io.neow3j.protocol.core.BlockParameterIndex;
+import io.neow3j.protocol.core.Request;
+import io.neow3j.protocol.core.methods.response.NeoBlock;
+import io.neow3j.protocol.core.methods.response.NeoBlockCount;
+import io.neow3j.protocol.core.methods.response.NeoGetBlock;
+import io.reactivex.Observable;
+import io.reactivex.disposables.Disposable;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.internal.matchers.apachecommons.ReflectionEquals;
+import org.mockito.stubbing.OngoingStubbing;
+
+public class JsonRpc2_0RxTest {
+
+    private Neow3j neow3j;
+
+    private Neow3jService neow3jService;
+
+    @Before
+    public void setUp() {
+        neow3jService = mock(Neow3jService.class);
+        neow3j = Neow3j.build(neow3jService, 1000, Executors.newSingleThreadScheduledExecutor());
+    }
+
+    @Test
+    public void testReplayBlocksObservable() throws Exception {
+
+        List<NeoGetBlock> neoGetBlocks = Arrays
+            .asList(createBlock(0), createBlock(1), createBlock(2));
+
+        OngoingStubbing<NeoGetBlock> stubbing =
+            when(neow3jService.send(any(Request.class), eq(NeoGetBlock.class)));
+        for (NeoGetBlock neoGetBlock : neoGetBlocks) {
+            stubbing = stubbing.thenReturn(neoGetBlock);
+        }
+
+        Observable<NeoGetBlock> observable = neow3j.replayBlocksObservable(
+            new BlockParameterIndex(BigInteger.ZERO),
+            new BlockParameterIndex(BigInteger.valueOf(2)),
+            false);
+
+        CountDownLatch transactionLatch = new CountDownLatch(neoGetBlocks.size());
+        CountDownLatch completedLatch = new CountDownLatch(1);
+
+        List<NeoGetBlock> results = new ArrayList<>(neoGetBlocks.size());
+        Disposable disposable = observable.subscribe(
+            result -> {
+                results.add(result);
+                transactionLatch.countDown();
+            },
+            throwable -> fail(throwable.getMessage()),
+            () -> completedLatch.countDown());
+
+        transactionLatch.await(1, TimeUnit.SECONDS);
+        assertThat(results, equalTo(neoGetBlocks));
+
+        disposable.dispose();
+
+        completedLatch.await(1, TimeUnit.SECONDS);
```

Comment:
Why do we have to wait for the completedLatch here? Is the completion notification only emitted after the subscription (Disposable) is disposed? Is the observable not already complete after emitting all its blocks?

Reply:
Addressed.